lib/backend/repository/repository1.go

```diff
@@ -0,0 +1,351 @@
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, -1, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != 200 {
+		return nil, -1, fmt.Errorf("HTTP code: %d, URL: %s", res.StatusCode, req.URL)
+	}
+
+	imageSize := int64(-1)
+
+	if hdr := res.Header.Get("X-Docker-Size"); hdr != "" {
+		imageSize, err = strconv.ParseInt(hdr, 10, 64)
+		if err != nil {
+			return nil, -1, err
+		}
+	}
+
+	b, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, -1, fmt.Errorf("failed to read downloaded json: %v (%s)", err, b)
+	}
+
+	return b, imageSize, nil
+}
+
+func (rb *RepositoryBackend) getLayerV1(imgID, registry string, repoData *RepoData, imgSize int64, tmpDir string) (*os.File, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", rb.protocol()+path.Join(registry, "images", imgID, "layer"), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	setAuthTokenV1(req, repoData.Tokens)
+	setCookieV1(req, repoData.Cookie)
+
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != 200 {
+		res.Body.Close()
+		return nil, fmt.Errorf("HTTP code: %d. URL: %s", res.StatusCode, req.URL)
+	}
+
+	// if we didn't receive the size via X-Docker-Size when we retrieved the
+	// layer's json, try Content-Length
+	if imgSize == -1 {
+		if hdr := res.Header.Get("Content-Length"); hdr != "" {
+			imgSize, err = strconv.ParseInt(hdr, 10, 64)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	prefix := "Downloading " + imgID[:12]
+	fmtBytesSize := 18
+	barSize := int64(80 - len(prefix) - fmtBytesSize)
+	bar := ioprogress.DrawTextFormatBarForW(barSize, os.Stderr)
+	fmtfunc := func(progress, total int64) string {
+		return fmt.Sprintf(
+			"%s: %s %s",
+			prefix,
+			bar(progress, total),
+			ioprogress.DrawTextFormatBytes(progress, total),
+		)
+	}
+
+	progressReader := &ioprogress.Reader{
+		Reader:       res.Body,
+		Size:         imgSize,
+		DrawFunc:     ioprogress.DrawTerminalf(os.Stderr, fmtfunc),
+		DrawInterval: 500 * time.Millisecond,
+	}
+
+	layerFile, err := ioutil.TempFile(tmpDir, "dockerlayer-")
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = io.Copy(layerFile, progressReader)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := layerFile.Sync(); err != nil {
+		return nil, err
+	}
+
+	return layerFile, nil
+}
+
+func setAuthTokenV1(req *http.Request, token []string) {
+	if req.Header.Get("Authorization") == "" {
+		req.Header.Set("Authorization", "Token "+strings.Join(token, ","))
+	}
+}
+
+func setCookieV1(req *http.Request, cookie []string) {
+	if req.Header.Get("Cookie") == "" {
+		req.Header.Set("Cookie", strings.Join(cookie, ""))
+	}
+}
+
+func makeEndpointsListV1(headers []string) []string {
+	var endpoints []string
+
+	for _, ep := range headers {
+		endpointsList := strings.Split(ep, ",")
+		for _, endpointEl := range endpointsList {
+			endpoints = append(
+				endpoints,
+				// TODO(iaguis) discover if httpsOrHTTP
```

Comment:
I guess this comment doesn't make sense anymore since we're using the `--insecure` flag to make it `http` or `https`